praw/models/reddit/removal_reasons.py

```diff
@@ -32,25 +33,17 @@ class RemovalReason(RedditBase):
 
     STR_FIELD = "id"
 
-    def __eq__(self, other: Union[str, _RemovalReason]) -> bool:
-        """Return whether the other instance equals the current."""
-        if isinstance(other, str):
-            return other == str(self)
-        return isinstance(other, self.__class__) and str(self) == str(other)
-
-    def __hash__(self) -> int:
-        """Return the hash of the current instance."""
-        return hash(self.__class__.__name__) ^ hash(str(self))
-
```

Comment:
Why did you remove these methods? Isn't this a breaking change?

Reply:
The methods are mostly identical to the RedditBase methods.

Reply:
The only difference is that lowercase and uppercase values of the same string are now equal, but that should not be a significant change.

Reply:
I'll revert the behavior and instead introduce it during PRAW 8.